File: android/src/main/java/com/dabchy/plugins/facebookevents/FacebookEventsPlugin.java
package com.dabchy.plugins.facebookevents;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "FacebookEvents")
public class FacebookEventsPlugin extends Plugin {

    private FacebookEvents facebookEvents;

    @Override
    public void load() {
        facebookEvents = new FacebookEvents(this);
    }

    @Override
    protected void handleOnStart() {
        super.handleOnStart();
        AppEventsLogger.activateApp(getActivity().getApplication());
    }

    @PluginMethod
    public void setAdvertiserTrackingEnabled(PluginCall call) {
        call.resolve();
    }

    @PluginMethod
    public void logEvent(PluginCall call) {
        String event = call.getString("event");
        JSObject params = call.getObject("params", new JSObject());
        facebookEvents.logEvent(event, params);
        call.resolve();
    }
}

File: ios/Plugin/FacebookEventsPlugin.swift
import Foundation
import Capacitor

@objc(FacebookEventsPlugin)
public class FacebookEventsPlugin: CAPPlugin {
    private let facebookEvents = FacebookEvents()

    @objc func setAdvertiserTrackingEnabled(_ call: CAPPluginCall) {
        guard let enabled = call.getBool("enabled") else {
            call.reject("Missing enabled argument")
            return
        }

        facebookEvents.setAdvertiserTrackingEnabled(enabled: enabled)
        call.resolve()
    }

    @objc func logEvent(_ call: CAPPluginCall) {
        guard let event = call.getString("event") else {
            call.reject("Missing event argument")
            return
        }

        let params = call.getObject("params") ?? [:]
        facebookEvents.logEvent(event: event, params: params as NSDictionary)
        call.resolve()
    }
}
